sql/src/main/java/io/crate/operation/projectors/BatchConsumerToRowReceiver.java

```diff
@@ -149,35 +144,11 @@ private RepeatHandle getRepeatHandle(BatchIterator iterator) {
 
     @Override
     public synchronized void kill(@Nullable Throwable throwable) {
-        killable.kill(throwable);
-    }
-
-    private static class KillableBatchIterator extends ForwardingBatchIterator implements Killable {
-
-        private final BatchIterator delegate;
-        private volatile Throwable killed;
-
-        KillableBatchIterator(BatchIterator delegate) {
-            this.delegate = delegate;
-        }
-
-        @Override
-        protected BatchIterator delegate() {
-            return delegate;
-        }
-
-        @Override
-        public boolean moveNext() {
-            if (killed == null) {
-                return super.moveNext();
-            }
-            Exceptions.rethrowUnchecked(killed);
-            return false;
-        }
-
-        @Override
-        public void kill(@Nullable Throwable throwable) {
-            killed = throwable;
+        rowReceiver.kill(throwable);
+        if (!running) {
+            // some RowReceiver kill implementations only set a internal "stop" flag and wait for the next
+            // row to process it - in order for that to work it's necessary to trigger the consumption
```

Comment:
even though somehow implicit, i think we should make it clear here that this is done only for bridging, maybe we should introduce some marker like XBridging?